statics/js/components/graph-layout.js

```diff
@@ -1,6 +1,184 @@
+const maxClockSkewMillis = 5 * 60 * 1000; // 5 minutes
+
+var LinkLabelBandwidth = Vue.extend({
+  mixins: [apiMixin],
+
+  methods: {
+
+    bandwidthFromMetrics: function(metrics) {
+      if (!metrics) {
+        return 0;
+      }
+      if (!metrics.Last) {
+        return 0;
+      }
+      if (!metrics.Start) {
+        return 0;
+      }
+
+      const totalByte = (metrics.RxBytes || 0) + (metrics.TxBytes || 0);
+      const deltaMillis = metrics.Last - metrics.Start;
+      const elapsedMillis = Date.now() - new Date(metrics.Last);
+
+      if (deltaMillis === 0) {
+        return 0;
+      }
+      if (elapsedMillis > maxClockSkewMillis) {
+        return 0;
+      }
+
+      return Math.floor(8 * totalByte * 1000 / deltaMillis); // bits-per-second 
+    },
+
+    setup: function(topology) {
+      this.topology = topology;
+    },
+
+    updateData: function(link) {
+      if (!link.target.metadata.LastUpdateMetric) {
+        return;
+      }
+
+      const defaultBandwidthBaseline = 1024 * 1024 * 1024; // 1 gbps
+
+      link.bandwidthBaseline = (this.topology.bandwidth.bandwidthThreshold === 'relative') ?
+        link.target.metadata.Speed || defaultBandwidthBaseline : 1;
+
+      link.bandwidthAbsolute = this.bandwidthFromMetrics(link.target.metadata.LastUpdateMetric);
+      link.bandwidth = link.bandwidthAbsolute / link.bandwidthBaseline;
+    },
+ 
+    hasData: function(link) {
+      if (!link.target.metadata.LastUpdateMetric) {
+        return false;
+      }
+
+      if (!link.bandwidth) {
+        return false;
+      }
+      return link.bandwidth > this.topology.bandwidth.active;
+    },
+
+    getText: function(link) {
+      return bandwidthToString(link.bandwidthAbsolute);
+    },
+
+    isActive: function(link) {
+      return (link.bandwidth > this.topology.bandwidth.active) && (link.bandwidth < this.topology.bandwidth.warning);
+    },
+
+    isWarning: function(link) {
+      return (link.bandwidth >= this.topology.bandwidth.warning) && (link.bandwidth < this.topology.bandwidth.alert);
+    },
+
+    isAlert: function(link) {
+      return link.bandwidth >= this.topology.bandwidth.alert;
+    },
+  },
+});
+
+var LinkLabelLatency = Vue.extend({
+  mixins: [apiMixin],
+
+  methods: {
+    setup: function(topology) {
+      this.active = 0;
+      this.warning = 10
+      this.alert = 100;
+    },
+
+    flowKey: function(a) {
+      return `TrackingID:${a.TrackingID},ANodeTID:${a.ANodeTID},BNodeTID:${a.BNodeTID}`;
+    },
+
+    flowCompare: function(a, b) {
+      const keya = this.flowKey(a);
+      const keyb = this.flowKey(b);
+      return keya.localeCompare(keyb);
```

Comment:
keya == keyb

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare
returns 0, <0 or >0 as needed by for the compare function.

The `==` operator returns `true` or `false`